pkg/build/ignore/docker/dockign_test.go

```diff
@@ -0,0 +1,112 @@
+package docker
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/openshift/source-to-image/pkg/api"
+	"os"
+)
+
+func newIgnorer() *DockIgn {
+	i := &DockIgn{
+	}
+	return i
+}
+
+func baseTest(t *testing.T, patterns []string, filesToCheck []string) {
+	cleanerr := os.RemoveAll("/tmp/upload")
+	if cleanerr != nil {
+		t.Errorf("problem cleaning up %v \n", cleanerr)
+	}
+	
+	c := &api.Config{WorkingDir: "/tmp"}
+	// create temp dir for .s2iignore that matches where dockin.go looks
+	dpath := filepath.Join(c.WorkingDir,"upload")
+	derr := os.Mkdir(dpath,0777)	
+	if derr != nil {
+		t.Errorf("Problem creating test dir %s with  %v \n", dpath, derr)
+	}
+	dpath = filepath.Join(dpath,"src")
+	derr = os.Mkdir(dpath,0777)
+	if derr != nil {
+		t.Errorf("Problem creating test dir %s with  %v \n", dpath, derr)
+	}
+	c.WorkingSourceDir = dpath
+	t.Logf("working source dir %s \n", dpath)
+
+	// create s2iignore file
+	ipath := filepath.Join(dpath, api.DefaultIgnoreFile)
+	ifile, ierr := os.Create(ipath)
+	defer ifile.Close()
+	if ierr != nil {
+		t.Errorf("Problem creating .s2iignore at %s with  %v \n",ipath, ierr)
+	}
```

Comment:
even better than doing all this might be to create a fixtures dir with the entire directory structure (including .s2iignore) that you want.  then you could just cp -r the fixtures dir and run, rather than programmatically constructing the directory structure.